x/budget/spec/04_begin_block.md

```diff
@@ -2,8 +2,17 @@
 
 # Begin-Block
 
-1. if current height % `params.EpochBlocks` != 0 or `params.EpochBlocks` == 0, exit.
-1. Get all the Budgets registered in params.Budgets and select only the valid budgets. If there is no valid budget, exit.
+At each `BeginBlock`, get all budgets registered in `params.Budgets` and select the valid budgets to distribute rewards for the previous block by its respective rate.
```

Comment:
```suggestion
At each `BeginBlock`, get all budgets registered in `params.Budgets` and select the valid budgets to collect budgets for the block by its respective rate.
```

Reply:
1. We should also fix [this comment](https://github.com/tendermint/budget/blob/b1fd56200d/x/budget/abci.go#L15)?
2. I think it should explain more about what it exactly does. If you look at this line [sendCoins](https://github.com/tendermint/budget/blob/b1fd56200d0b34708c9384ee725b8d83cb6649ca/x/budget/keeper/budget.go#L57), it sends `collectedCoins` from `budgetSourceAcc ` to `collectionAcc`. Maybe we should add a context about this?

Reply:
1. Yes, I think the comment was a mistake while copying the distribution module code.
2. Yes, That would be better.